modules/gateway/persist.go

```diff
@@ -1,38 +1,30 @@
 package gateway
 
 import (
-	"encoding/json"
 	"log"
 	"os"
 	"path/filepath"
 
 	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/persist"
 )
 
+var persistMetadata = persist.Metadata{
+	Header:  "Sia Node List",
+	Version: "0.3.3",
+}
+
 func (g *Gateway) save() error {
 	var nodes []modules.NetAddress
 	for node := range g.nodes {
 		nodes = append(nodes, node)
 	}
-	data, err := json.MarshalIndent(nodes, "", "\t")
-	if err != nil {
-		return err
-	}
-	file, err := os.Create(filepath.Join(g.saveDir, "nodes.json"))
-	if err != nil {
-		return err
-	}
-	_, err = file.Write(data)
-	return err
+	return persist.SaveFile(persistMetadata, nodes, filepath.Join(g.saveDir, "nodes.json"))
 }
 
 func (g *Gateway) load() error {
-	file, err := os.Open(filepath.Join(g.saveDir, "nodes.json"))
-	if err != nil {
-		return err
-	}
 	var nodes []modules.NetAddress
-	err = json.NewDecoder(file).Decode(&nodes)
+	err := persist.LoadFile(persistMetadata, &nodes, filepath.Join(g.saveDir, "nodes.json"))
```

Comment:
That gives me an idea. Instead of putting everything in one file, we can put each major type of thing into a different file. That makes it relatively easy for users to swap out and modify certain configurations without needing to worry about preserving completely unrelated stuff.

future thoughts, of course.